lint/src/main/java/permissions/dispatcher/CallNeedsPermissionDetector.java

```diff
@@ -1,54 +1,80 @@
 package permissions.dispatcher;
 
 
-import com.android.tools.lint.client.api.JavaParser;
 import com.android.tools.lint.detector.api.Category;
 import com.android.tools.lint.detector.api.Detector;
 import com.android.tools.lint.detector.api.Implementation;
 import com.android.tools.lint.detector.api.Issue;
 import com.android.tools.lint.detector.api.JavaContext;
 import com.android.tools.lint.detector.api.Scope;
 import com.android.tools.lint.detector.api.Severity;
+import com.intellij.psi.JavaElementVisitor;
+import com.intellij.psi.PsiAnnotation;
+import com.intellij.psi.PsiClass;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiMethod;
+import com.intellij.psi.PsiMethodCallExpression;
+import com.intellij.psi.PsiReferenceExpression;
 
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.EnumSet;
 import java.util.HashSet;
 import java.util.List;
 import java.util.Set;
 
-import lombok.ast.Annotation;
-import lombok.ast.AstVisitor;
-import lombok.ast.ClassDeclaration;
-import lombok.ast.ForwardingAstVisitor;
-import lombok.ast.MethodInvocation;
 
+public class CallNeedsPermissionDetector extends Detector implements Detector.JavaPsiScanner {
 
-public class CallNeedsPermissionDetector extends Detector implements Detector.JavaScanner {
     public static final Issue ISSUE = Issue.create("CallNeedsPermission",
             "Call the corresponding \"withCheck\" method of the generated PermissionsDispatcher class instead",
             "Directly invoking a method annotated with @NeedsPermission may lead to misleading behaviour on devices running Marshmallow and up. Therefore, it is advised to use the generated PermissionsDispatcher class instead, which provides a \"withCheck\" method that safely handles runtime permissions.",
             Category.CORRECTNESS,
             7,
             Severity.ERROR,
-            new Implementation(CallNeedsPermissionDetector.class, EnumSet.of(Scope.ALL_JAVA_FILES)));
+            new Implementation(CallNeedsPermissionDetector.class,
+                    EnumSet.of(Scope.ALL_JAVA_FILES)));
 
     static List<String> generatedClassNames = new ArrayList<String>();
 
+    static List<String> METHODS = Collections.emptyList();
```

Comment:
Well, it's a little bit not intuitive to add something to final list for me, so how about make this variable being just static and rename like normal variable?